#160451521 user should be able to track comments history

kampkelly opened the pull request:
#### What does this PR do?
This pull request adds keeping records of comments history

#### Description of Task to be completed?
- Write tests for tracking comment history
- Write the code to track comment history

#### How should this be manually tested?
Using postman and as a signed in user, make requests to the following api endpoints.

To update a comment: /api/v1/articles/:articleSlug/comments/commentId/edit -> PUT request (params: comment)
To retrieve comment history for a user: /api/v1/articles/:articleSlug/comments/:commentId/myhistory -> GET request
To retrieve comment history for all users: /api/v1/articles/:articleSlug/comments/:commentId/allhistory -> GET request

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#160451521
#### Questions:```
None

kampkelly committed:
feature(comments-history): track comments history

This commit adds the functionality to see previous versions of a comment.
The tests for this functionality have also been added.

Description: When a user edits any of his comments, then the current comment before it is updated should be kept for record purposes.

Finishes #160451521

kampkelly committed:
feature(comments-history): refactor code

This commit refactors the commentsHistoryController. Reused code has been put in a helper function.

Finishes #160451521

kampkelly commented on server/controllers/commentHistoryController.js:
```diff
@@ -0,0 +1,70 @@
+        const searchParams = {
+          commentId: req.params.commentId, articleSlug: req.params.articleSlug, userId: req.user.id
+        };
+        getCommentHistory(searchParams, res, 'comment history retrieved', 'you have no histories for this comment');
```
This is calling a helper function. The same block of code was used in multiple functions in this controller. So I moved it to a helper function. The strings sent along are the specific response messages for this function.

kampkelly commented on server/controllers/commentHistoryController.js:
```diff
@@ -0,0 +1,70 @@
+        const searchParams = {
+          commentId: req.params.commentId, articleSlug: req.params.articleSlug
+        };
+        getCommentHistory(searchParams, res, 'all comment histories retrieved', 'there are no histories for this comment');
```
This is calling a helper function. The same block of code was used in multiple functions in this controller. So I moved it to a helper function. The strings sent along are the specific response messages for this function.

kampkelly committed:
feature(comments-history): disable eslint on isNaN

This commit disables eslint checkinmg isNaN in articles helper.

slimsolz reviewed (commented):
LGTM

kampkelly commented on server/helpers/article.js:
```diff
@@ -1,6 +1,7 @@
 
 export const computeOffset = (req) => {
   const page = Object.keys(req.query).length ? parseInt(req.query.page, 10) : 1;
+  // eslint-disable-next-line
```
Eslint is throwing an error on isNaN ---> 'no-restricted-globals'. So I disabled the check for this line.

samdiano reviewed (commented):
LGTM

s-akinrele commented on server/tests/commentsHistory.models.test.js:
```diff
@@ -0,0 +1,98 @@
+import db from '../models';
+
+
+const validCommentHistory = {
```
move it to another file

s-akinrele commented on server/tests/commentsHistory.test.js:
```diff
@@ -0,0 +1,90 @@
+chai.use(chaiHttp);
+
+const user = {
+  username: faker.internet.userName(),
```
and this

s-akinrele reviewed (changes requested):
I had some comments

kampkelly committed:
feature(comments-history): refactor tests

This commit refactors the tests. The constant variables have beenm moved to the helper file.

kampkelly committed:
Merge branch 'staging' of github.com:andela/phlox-ah into feature/160451521/track-comment-history

# Conflicts:
#	server/controllers/commentController.js
#	server/helpers/article.js
